tt_metal/api/program.hpp

```diff
@@ -0,0 +1,128 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "types.hpp"
+
+#include "tt_metal/impl/kernels/kernel_types.hpp"
+#include "tt_metal/impl/buffers/circular_buffer_types.hpp"
+
+//==================================================
+//                  PROGRAM MANAGEMENT
+//==================================================
+
+namespace tt::tt_metal{
+namespace v1 {
+
+/**
+ * @brief Creates a Program object, which bundles kernels, circular buffers, and semaphores for execution on the device.
+ *
+ * @return Program handle to the created program.
+ */
+Program CreateProgram();
+
+/**
+ * @brief Creates a data movement or compute kernel and adds it to the program.
+ *
+ * @param program The program to which this kernel will be added.
+ * @param file_name Path to the kernel source file.
+ * @param core_spec Specifies the cores on which the kernel will be placed (CoreCoord, CoreRange, or CoreRangeSet).
+ * @param config Configuration for the kernel (DataMovementConfig, ComputeConfig, or EthernetConfig).
+ * @return KernelHandle representing the kernel ID.
+ */
+KernelHandle CreateKernel(
+    Program program,
+    const std::string &file_name,
+    const CoreRangeSet &core_spec,
+    const std::variant<DataMovementConfig, ComputeConfig, EthernetConfig> &config);
```

Comment:
These should be introduced with separate overloads rather than type-erased into a variant

Reply:
done